purpose.tex

```diff
@@ -5,13 +5,34 @@ \section{Purpose}
+  \end{description}
+\item[Metadata markup] A PAML protocol description (possibly with some ancillary information supplied by, e.g., an OPIL document, the container ontology, etc.) should support automatically tagging with metadata data that are collected in the course of protocol execution, \emph{to the extent lab automation at the executing lab supports this}.  For example, a protocol that collects flow cytometry data on samples of different strains under varying growth conditions, should support automatically recording the strain, growth conditions, time of data collection, calibration, etc. of these measurements.
+\item[Planning and scheduling] Lab resources are valuable and scarce, and some organizations will want to be able to optimize their use. To do so, PAML should support (at least) extraction of resource requirements from activities in the protocol, and estimated durations.  Note that \emph{what} resource requirements and duration estimates are required will likely be a function of both the protocol and the lab.  What resources are limited, and must be considered in a planner or scheduler, versus those that can be effectively treated as unlimited, will vary by lab.  Management styles will also vary between organizations.
+\item[Verification] Authoring a formal protocol in PAML requires substantial mental effort, and the usefulness of the corresponding artifacts can be compromised if they are ill-formed or erroneous.  Supporting protocol authors in achieving correctness is an important goal. Some of this will fall onto the shoulders of those building PAML support tools, however the specification itself must provide guidance as to what it means for a protocol to be complete, consistent, etc.
+\item[Mapping protocols to labs] PAML aims to support replication in part by providing automated support for mapping protocols developed at one lab onto another.  We suspect that for some time to come this will be at best a semi-automated process, and like verification, this task will fall heavily on those who provide tool support for PAML users.  However, also as for verification, the PAML specification must say what it means to perform this task correctly -- what a protocol specification truly requires, what aspects can safely be varied, and which must be honored.  This task is closely tied to the task of verification.
+\item[``Derived works"] As with most complex formal artifacts (programs, spreadsheets, etc.), we confidently expect that few users will create new protocols from an entirely blank slate.  Instead, most will take an existing protocol, copy it, and edit the copy.  PAML must support this mode of operation.  Key issues here are which component structures can be incorporated by reference, and which must be copied instead.  Tracking the relationship between an original protocol, and protocols that have branched from it would also be desirable, in order to support, for example, propagating fixes from a root protocol to others.
```

Comment:
I think it would be good to also mention ongoing development and versioning of a protocol.

Reply:
Done.  One issue we might want to think about is when we are refining and existing protocol to make a new version, and when we are creating a derived protocol.

This is not going to be something that we can provide a hard, crisp answer to, but it is something about which we might wish to provide guidance.